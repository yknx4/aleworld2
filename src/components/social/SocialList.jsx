import React,{Component, PropTypes} from 'react';
import SocialLink from './SocialLink';

export default class SocialList extends Component {

  drawLink(item) {
    return <li key={item.icon}><SocialLink {...item} /></li>;
  }

  render() {
    const {items} = this.props;
    return (
      <ul className="list-inline social-list spread-children">
        {items.map(this.drawLink)}
      </ul>
    );
  }
}

SocialList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};
