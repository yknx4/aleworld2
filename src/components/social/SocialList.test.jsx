import React from 'react';
import { shallow } from 'enzyme';
import SocialList from './SocialList';

describe('SocialList', () => {
  let shallowComponent;
  const socialItems = [
    { icon: 'github', url: 'url1' },
    { icon: 'facebook', url: 'url2' },
  ];

  beforeEach(() => {
    const component = <SocialList items={socialItems}/>;
    shallowComponent = shallow(component);
  });

  it('should render without throwing an error', function() {
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should be selectable by tag "ul"', function() {
    expect(shallowComponent.is('ul')).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render 2 elements', () => {
    expect(shallowComponent.find('li').length).toBe(socialItems.length);
  });

});
