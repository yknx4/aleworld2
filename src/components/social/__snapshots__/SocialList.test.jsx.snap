// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialList should be selectable by tag "ul" 1`] = `
<ul
  className="list-inline social-list spread-children"
>
  <li>
    <SocialLink
      icon="github"
      target="blank"
      url="url1"
    />
  </li>
  <li>
    <SocialLink
      icon="facebook"
      target="blank"
      url="url2"
    />
  </li>
</ul>
`;

exports[`SocialList should render without throwing an error 1`] = `
<ul
  className="list-inline social-list spread-children"
>
  <li>
    <SocialLink
      icon="github"
      target="blank"
      url="url1"
    />
  </li>
  <li>
    <SocialLink
      icon="facebook"
      target="blank"
      url="url2"
    />
  </li>
</ul>
`;
